neo/framework/Common.cpp

```diff
@@ -246,12 +246,29 @@ class idCommonLocal : public idCommon {
 	idCompressor *				config_compressor;
 #endif
 
-	SDL_TimerID					async_timer;
+	static int					AsyncThread(void* arg);
+	xthreadInfo					asyncThread;
+	volatile bool				runAsyncThread;
```

Comment:
`volatile` is not recommended for multithreading. Prefer `std::atomic`.

https://stackoverflow.com/questions/2484980/why-is-volatile-not-considered-useful-in-multithreaded-c-or-c-programming

Reply:
That would require C++11, though.
I know this is not exactly clean, but I also don't think it can cause trouble: The only time that bool is changed while the thread is running is when it's set to `false` on shutdown.
Even if the modification is not atomic (however that should happen in a byte-sized bool), it doesn't matter if that thread runs one more frame than it technically should - eventually it will see the changed value (as `volatile` makes sure it's really re-read on each iteration).